pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -67,7 +67,8 @@ type APIResource struct {
 	// Lower-cased plural name of the resource (e.g. configmaps).  If
 	// not provided, it will be computed by lower-casing the kind and
 	// suffixing an 's'.
-	PluralName string `json:"pluralName,omitempty"`
+	PluralName string   `json:"pluralName,omitempty"`
+	ShortNames []string `json:"shortNames,omitempty"`
```

Comment:
This struct is only used in FederatedTypeConfig type, and only includes the fields required by a controller to access the api of a kubernetes type.  A short name is a UX nicety that controllers won't use, so specifying short names in this context does not make sense.